pkg/cloud/openstack/services/networking/loadbalancer.go

```diff
@@ -0,0 +1,437 @@
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func checkIfLbExists(networkingClient *gophercloud.ServiceClient, name string) (*loadbalancers.LoadBalancer, error) {
+	allPages, err := loadbalancers.List(networkingClient, loadbalancers.ListOpts{Name: name}).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	lbList, err := loadbalancers.ExtractLoadBalancers(allPages)
+	if err != nil {
+		return nil, err
+	}
+	if len(lbList) == 0 {
+		return nil, nil
+	}
+	return &lbList[0], nil
+}
+
+func checkIfFloatingIPExists(networkingClient *gophercloud.ServiceClient, ip string) (*floatingips.FloatingIP, error) {
+	allPages, err := floatingips.List(networkingClient, floatingips.ListOpts{FloatingIP: ip}).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	fpList, err := floatingips.ExtractFloatingIPs(allPages)
+	if err != nil {
+		return nil, err
+	}
+	if len(fpList) == 0 {
+		return nil, nil
+	}
+	return &fpList[0], nil
+}
+
+func checkIfListenerExists(networkingClient *gophercloud.ServiceClient, name string) (*listeners.Listener, error) {
+	allPages, err := listeners.List(networkingClient, listeners.ListOpts{Name: name}).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	listenerList, err := listeners.ExtractListeners(allPages)
+	if err != nil {
+		return nil, err
+	}
+	if len(listenerList) == 0 {
+		return nil, nil
+	}
+	return &listenerList[0], nil
+}
+
+func checkIfPoolExists(networkingClient *gophercloud.ServiceClient, name string) (*pools.Pool, error) {
+	allPages, err := pools.List(networkingClient, pools.ListOpts{Name: name}).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	poolList, err := pools.ExtractPools(allPages)
+	if err != nil {
+		return nil, err
+	}
+	if len(poolList) == 0 {
+		return nil, nil
+	}
+	return &poolList[0], nil
+}
+
+func checkIfMonitorExists(networkingClient *gophercloud.ServiceClient, name string) (*monitors.Monitor, error) {
+	allPages, err := monitors.List(networkingClient, monitors.ListOpts{Name: name}).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	monitorList, err := monitors.ExtractMonitors(allPages)
+	if err != nil {
+		return nil, err
+	}
+	if len(monitorList) == 0 {
+		return nil, nil
+	}
+	return &monitorList[0], nil
+}
+
+func checkIfLbMemberExists(networkingClient *gophercloud.ServiceClient, poolID, name string) (*pools.Member, error) {
+	allPages, err := pools.ListMembers(networkingClient, poolID, pools.ListMembersOpts{Name: name}).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	lbMemberList, err := pools.ExtractMembers(allPages)
+	if err != nil {
+		return nil, err
+	}
+	if len(lbMemberList) == 0 {
+		return nil, nil
+	}
+	return &lbMemberList[0], nil
+}
+
+var backoff = wait.Backoff{
+	Steps:    10,
+	Duration: 30 * time.Second,
+	Factor:   1.0,
+	Jitter:   0.1,
+}
+
+func waitForLoadBalancer(networkingClient *gophercloud.ServiceClient, id, target string) error {
+	klog.Infof("Waiting for loadbalancer %s to become %s.", id, target)
```

Comment:
What are all possible values of `target`? Link to documentation would be helpful

Reply:
Good point. I'll add this link: https://developer.openstack.org/api-ref/network/v2/?expanded=show-load-balancer-status-tree-detail#load-balancer-statuses

Reply:
fixed